x-pack/plugins/features/server/feature_registry.ts

```diff
@@ -61,6 +64,112 @@ export class FeatureRegistry {
     this.esFeatures[feature.id] = featureCopy;
   }
 
+  /**
+   * Updates definitions for the registered features using configuration overrides, if any.
+   */
+  public applyOverrides(overrides: ConfigOverridesType) {
+    for (const [featureId, featureOverride] of Object.entries(overrides)) {
+      const feature = this.kibanaFeatures[featureId];
+      if (!feature) {
+        throw new Error(
+          `Cannot override feature "${featureId}" since feature with such ID is not registered.`
+        );
+      }
+
+      if (featureOverride.disabled) {
+        feature.disabled = featureOverride.disabled;
+      }
+
+      // Note that the name doesn't currently support localizable strings. We'll revisit this approach when i18n support
+      // becomes necessary.
+      if (featureOverride.name) {
+        feature.name = featureOverride.name;
+      }
+
+      if (featureOverride.category) {
+        feature.category = DEFAULT_APP_CATEGORIES[featureOverride.category];
+      }
+
+      if (featureOverride.order != null) {
+        feature.order = featureOverride.order;
+      }
+
+      if (featureOverride.privileges) {
+        for (const [privilegeId, privilegeOverride] of Object.entries(featureOverride.privileges)) {
+          const typedPrivilegeId = privilegeId as 'read' | 'all';
+          const targetPrivilege = feature.privileges?.[typedPrivilegeId];
+          if (!targetPrivilege) {
+            throw new Error(
+              `Cannot override privilege "${privilegeId}" of feature "${featureId}" since "${privilegeId}" privilege is not registered.`
+            );
+          }
+
+          for (const featureReference of privilegeOverride.composedOf ?? []) {
+            const referencedFeature = this.kibanaFeatures[featureReference.feature];
+            if (!referencedFeature) {
+              throw new Error(
+                `Cannot compose privilege "${privilegeId}" of feature "${featureId}" with privileges of feature "${featureReference.feature}" since such feature is not registered.`
+              );
+            }
+
+            // Collect all known feature and sub-feature privileges for the referenced feature.
+            const knownPrivileges = new Map(
+              Object.entries(referencedFeature.privileges ?? {}).concat(
+                collectSubFeaturesPrivileges(referencedFeature)
+              )
+            );
+
+            for (const privilegeReference of featureReference.privileges) {
+              const referencedPrivilege = knownPrivileges.get(privilegeReference);
+              if (!referencedPrivilege) {
+                throw new Error(
+                  `Cannot compose privilege "${privilegeId}" of feature "${featureId}" with privilege "${privilegeReference}" of feature "${featureReference.feature}" since such privilege is not registered.`
+                );
+              }
+
+              if (referencedPrivilege.cases || referencedPrivilege.alerting) {
```

Comment:
Could you explain further if possible why privileges that include `alerting` cannot be composed? 

I was hoping to compose privileges for observability. For example, I was hoping to compose "APM" with Logs and Observability. These two features do have alert privileges attached, but I'd like those to come along with the composed privilege.  

cc: @kobelb

Reply:
I _think_ Oleg was trying to reduce scope & complexity by disallowing composition that [we believed] wasn't required for MVP. I can't speak to technical limitations one way or the other, so we'll unfortunately need @azasypkin to weigh in when he returns.

Reply:
> I think this is something we need to explore in this PR, rather than postponing, unfortunately, from the Observability perspective.

Roger that. Would you mind updating the `Observability Feature Toggle Proposal` with the exact details of the changes you'd like to see (what privileges should be hidden, and what other existing privileges should implicitly grant those privileges instead)? I'll tag you on that document.

> Alternatively if there was a way to automatically "bestow" Logs and Observability privileges onto all serverless users for the observability project type, I think we might take that as well.

No, unfortunately, our authorization model doesn't support anything like that yet.